camel/storages/graph_storages/neo4j_graph.py

```diff
@@ -0,0 +1,547 @@
+
+    def _delete_rel(self, subj: str, obj: str, rel: str) -> None:
+        r"""Deletes a specific relationship between two nodes in the Neo4j
+        database.
+
+        Args:
+            subj (str): The identifier for the subject entity.
+            obj (str): The identifier for the object entity.
+            rel (str): The relationship between the subject and object to
+                delete.
+        """
+        with self.driver.session(database=self.database) as session:
+            session.run(
+                ("MATCH (n1:{})-[r:{}]->(n2:{}) WHERE n1.id = $subj AND n2.id"
+                 " = $obj DELETE r").format(BASE_ENTITY_LABEL.replace("_", ""),
+                                            rel,
+                                            BASE_ENTITY_LABEL.replace("_",
+                                                                      "")),
+                {
+                    "subj": subj,
+                    "obj": obj
+                },
+            )
+
+    def _delete_entity(self, entity: str) -> None:
+        r"""Deletes an entity from the Neo4j database based on its unique
+        identifier.
+
+        Args:
+            entity (str): The unique identifier of the entity to be deleted.
+        """
+        with self.driver.session(database=self.database) as session:
+            session.run(
+                "MATCH (n:%s) WHERE n.id = $entity DELETE n" %
+                BASE_ENTITY_LABEL.replace("_", ""),
+                {"entity": entity},
+            )
+
+    def _check_edges(self, entity: str) -> bool:
+        r"""Checks if the given entity has any relationships in the graph
+        database.
+
+        Args:
+            entity (str): The unique identifier of the entity to check.
+
+        Returns:
+            bool: True if the entity has at least one edge (relationship),
+                False otherwise.
+        """
+        with self.driver.session(database=self.database) as session:
+            is_exists_result = session.run(
+                "MATCH (n1:%s)--() WHERE n1.id = $entity RETURN count(*)" %
+                (BASE_ENTITY_LABEL.replace("_", "")),
+                {"entity": entity},
+            )
+            return bool(list(is_exists_result))
+
+    def delete_triplet(self, subj: str, obj: str, rel: str) -> None:
+        r"""Deletes a specific triplet from the graph, comprising a subject,
+        object and relationship.
+
+        Args:
+            subj (str): The identifier for the subject entity.
+            obj (str): The identifier for the object entity.
+            rel (str): The relationship between the subject and object.
+        """
+        self._delete_rel(subj, obj, rel)
+        if not self._check_edges(subj):
+            self._delete_entity(subj)
+        if not self._check_edges(obj):
+            self._delete_entity(obj)
+
+    def _get_node_import_query(self, base_entity_label: bool,
```

Comment:
What's the functionality of this function? Basically what does node import mean?

Reply:
write nodes into Neo4j, including create&merge nodes, add label(base_entity_label), add property (include_source)